File: Examples/DotNET/SDK/VisualBasic/Mail-Merge/WithoutRegionsWithoutCloudStorage.vb
Imports Aspose.Cloud

Namespace Mail_Merge
    Class WithoutRegionsWithoutCloudStorage
        Public Shared Sub Run()
            Dim subdirectory As String = "MailMerge"
            Dim inputfile As String = "WithoutRegionsWithoutStorage.doc"
            Dim xmlfile As String = "WithoutRegionsWithoutStorage.xml"

            ' The path to the documents directory.
            Dim dataDir As String = (Common.GetDataDir() & subdirectory) + "\"

            ' Upload input file from local directory to Cloud Storage
            Common.UploadFile(inputfile, subdirectory)

            Dim strXML As String = String.Empty
            Using sr As New System.IO.StreamReader(dataDir & xmlfile)
                strXML = sr.ReadToEnd()
            End Using

            Common.WordsService.ExecuteDocumentMailMergeOnline(strXML, False, String.Empty, inputfile, Common.GetOutputFilePath(inputfile, True), Common.FOLDER)

            ' Download output file from cloud storage and save on local directory
            dataDir = Common.DownloadFile(inputfile, subdirectory)

            Console.WriteLine(Convert.ToString(vbLf & "Mail merge without regions and without cloud storage executed successfully." & vbLf & "File saved at ") & dataDir)
        End Sub
    End Class
End Namespace

File: Examples/DotNET/SDK/CSharp/Mail-Merge/WithoutRegionsWithoutCloudStorage.cs
using System;
using Aspose.Cloud;

namespace CSharp.Mail_Merge
{
    class WithoutRegionsWithoutCloudStorage
    {
        public static void Run()
        {
            string subdirectory = "MailMerge";
            string inputfile = "WithoutRegionsWithoutStorage.doc";
            string xmlfile = "WithoutRegionsWithoutStorage.xml";
            
            // The path to the documents directory.
            string dataDir = Common.GetDataDir() + subdirectory + "\\";
           
            // Upload input file from local directory to Cloud Storage
            Common.UploadFile(inputfile, subdirectory);

            string strXML = string.Empty;
            using (System.IO.StreamReader sr = new System.IO.StreamReader(dataDir + xmlfile))
            {
                strXML = sr.ReadToEnd();
            }

            Common.WordsService.ExecuteDocumentMailMergeOnline(strXML, false, string.Empty, inputfile, Common.GetOutputFilePath(inputfile, true), Common.FOLDER);

            // Download output file from cloud storage and save on local directory
            dataDir = Common.DownloadFile(inputfile, subdirectory);

            Console.WriteLine("\nMail merge without regions and without cloud storage executed successfully.\nFile saved at " + dataDir);
        }
    }
}
